test/net/sourceforge/kolmafia/utilities/StringUtilitiesTest.java

```diff
@@ -13,8 +15,292 @@ void isVowel() {
+    String testMe = buildTestString(159);
+    testMe = manuallyInsertBreak(testMe, 40);
+    String expected = manuallyInsertBreak(testMe, 80);
+    expected = expected + BREAK;
+    assertEquals(expected, StringUtilities.basicTextWrap(testMe), "Breaks not where expected.");
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "'1234', '1234'",
+    "'12345', '12345\n'",
+    "'1234567890', '12345\n67890\n'",
+    "'123456789012', '12345\n67890\n12'",
+    "'123\n4567890123456789', '123\n4\n56789\n01234\n56789\n'",
+    "'12 45', '12\n45'",
+    "'1 3 5 7 9 1', '1 3 5\n7 9\n1'",
+    "'1 2  \n  4 5\n 6 \n', '1 2\n4 5\n6'",
+    "'1234 67', '1234\n67'"
+  })
+  public void exerciseTextWrapWithShortWrapLength(String input, String expected) {
+    assertEquals(expected, StringUtilities.basicTextWrap(input, 5));
+  }
+
+  /*
+  This builds a string of the specified length.  It uses the repeating pattern "1234567891" to
+  make it easier for humans to debug.  The modulo 10 is just to get a single digit as a string;
+   */
+  private String buildTestString(int length) {
+    StringBuilder retVal = new StringBuilder();
+    for (int i = 1; i <= length; i++) {
+      retVal.append(i % 10);
+    }
+    return retVal.toString();
+  }
+  /*
+  This inserts a break into a string
+   */
+  private String manuallyInsertBreak(String input, int breakPos) {
+    if (breakPos >= input.length()) {
+      return input;
+    } else {
+      return input.subSequence(0, breakPos) + BREAK + input.subSequence(breakPos, input.length());
+    }
+  }
+  // End tests for basicTestWrap
+
+  @ParameterizedTest
+  @CsvSource({
+    "'not an id', '-1'",
+    "'[not an id]', '-1'",
+    "'123','-1'",
+    "'[123', '-1'",
+    "'123[', '-1'",
+    "'123', '-1'",
+    "'[1337]2468 googles', '1337'",
+    "'[123]unreal item', '123'"
+  })
+  public void itShouldExerciseGetBracketedID(String name, String id) {
+    assertEquals(Integer.parseInt(id), StringUtilities.getBracketedId(name));
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "'not an id', 'not an id'",
+    "'[not an id]', '[not an id]'",
+    "'[123', '[123'",
+    "'123]', '123]'",
+    "'123', '123'",
+    "'[123]unreal item', 'unreal item'",
+    "'[1337]2468 goggles', '2468 goggles'"
+  })
+  public void itShouldExerciseRemoveBracketedID(String name, String id) {
+    assertEquals(id, StringUtilities.removeBracketedId(name));
+  }
+
+  @Test
+  public void itShouldTokenizeStringsWithDefaults() {
+    // simple comma separated list
+    String testString = "folder (cyan), folder (magenta), folder (yellow)";
+    List<String> expected = new ArrayList<>();
+    expected.add("folder (cyan)");
+    expected.add("folder (magenta)");
+    expected.add("folder (yellow)");
+    try {
+      List<String> result = StringUtilities.tokenizeString(testString);
+      assertEquals(expected, result, "Lists are not the same.");
+    } catch (Exception e) {
+      fail(e.getMessage());
```

Comment:
I think if a method throws the test will fail by default. Not sure, but I think that would simplify things?

Reply:
One test case was designed to trigger the exception and a thrown exception means the test passed.  An exception in the other cases was a test failure.  I was trying to make that distinction clear in the code.  Since I had packed distinct cases in one test I also did not want an unexpected exception to be a failure.